app/models/profile.rb

```diff
@@ -18,12 +18,26 @@ class Profile < ApplicationRecord
 
   attr_reader :personal_key
 
+  # Rails natively handles JSONB columns, however we have been double-serializing our data
+  # until now (saving JSON objects as JSON atoms)
+  # We can remove this override once run the backfill job to re-encode older values
+  def proofing_components
+    value = super
+    if value.present?
+      if value.kind_of?(Hash)
+        value
+      elsif value.kind_of?(String)
+        JSON.parse(value)
+      end
+    end
+  end
+
   # rubocop:disable Rails/SkipsModelValidations
   def activate
     now = Time.zone.now
     is_reproof = Profile.find_by(user_id: user_id, active: true)
     transaction do
-      Profile.where('user_id=?', user_id).update_all(active: false)
+      Profile.where(user_id:  user_id).update_all(active: false)
```

Comment:
technically this has nothing to do with the rest of the PR but I figured I'd clean it up while we're here, happy to undo the change if anybody feels strongly